18F/identity-idp
# LG-13054: Remove automatic account creation for forgot password missing user

## 🎫 Ticket

[LG-13054](https://cm-jira.usa.gov/browse/LG-13054)

## 🛠 Summary of changes

Updates forgotten password flow to avoid creating a stub account when submitting an email that is not associated with a user, to prevent waste in the likely scenario that someone may not complete the process to create an account. Instead, links directly to the standard "Create an account" screen.

In doing so, this also addresses a few additional issues:

- The existing email does not preserve the `request_id` of the service provider if a user were to click the "Try a different email address" primary link in the email, and therefore they would likely not arrive back at the service provider as intended
- Ensures that the user has the opportunity to select their preferred communication language preference

## 📜 Testing Plan

Verify that forgotten password for missing account directs new account creation to "Create an account" page:

1. Go to http://localhost:3000
2. Click "Forgot your password?
3. Enter an email address for an account that does not exist
4. Click "Continue"
5. Observe email "Email not found" is shown
6. Click "create a new account" link in the email body
7. Observe that you're directed to the "Create an account for new users" page

Verify that `request_id` is retained in "Try a different email address" link:

1. Go to http://localhost:3000/rails/mailers/anonymous_mailer/password_reset_missing_user
8. Click "Try a different email address"
9. Observe `?request_id=` parameter in URL

## 👀 Screenshots

Before|After
---|---
![image](https://github.com/18F/identity-idp/assets/1779930/1cdea695-24f1-4d21-926d-22915e686507)|![image](https://github.com/18F/identity-idp/assets/1779930/054716fc-5541-4908-aa8c-351064ba3126)
